nlogger/structured.go

```diff
@@ -101,7 +98,16 @@ func (s *provider) Get() Structured {
 
 // Replace the logger inside the Provider
 func (s *provider) Replace(l Structured) {
-	s.mut.Lock()
+	defer func() {
+		// will handle panic only if it is caused by a non-nil interface
+		if l != nil {
+			if r := recover(); r != nil {
+				var logger = s.Get().(Structured)
+				logger.Error(
+					"your new logger is not of the same concrete type of logger as your old logger, " +
```

Comment:
Can you get the types of both and include it in the error message? I'm ok with using reflection in here